includes/reader-activation/class-reader-data.php

```diff
@@ -31,6 +31,13 @@ public static function init() {
 		add_action( 'rest_api_init', [ __CLASS__, 'register_routes' ] );
 		add_action( 'wp', [ __CLASS__, 'setup_reader_activity' ] );
 		add_action( 'wp_enqueue_scripts', [ __CLASS__, 'config_script' ] );
+
+		/* Update reader data items on event dispatches */
+		add_action( 'newspack_data_event_dispatch_newsletter_subscribed', [ __CLASS__, 'set_is_newsletter_subscriber' ], 10, 2 );
+		add_action( 'newspack_data_event_dispatch_newsletter_updated', [ __CLASS__, 'set_is_newsletter_subscriber' ], 10, 2 );
+		add_action( 'newspack_data_event_dispatch_donation_new', [ __CLASS__, 'set_is_donor' ], 10, 2 );
+		add_action( 'newspack_data_event_dispatch_donation_subscription_new', [ __CLASS__, 'set_is_donor' ], 10, 2 );
```

Comment:
We don't need this. `donation_new` fires for both one time and recurring donations

Reply:
Updated in e808ad5800d78bfe22769291da7cded16dc4d4e8